usecases/Concrete/simulation_model/material_problems/linear_elasticity.py

```diff
@@ -0,0 +1,108 @@
+warnings.simplefilter("ignore", QuadratureRepresentationDeprecationWarning)
+
+
+# full concrete model, including hydration-temperate and mechanics, including calls to solve etc.
+class LinearElasticity(MaterialProblem):
+    def __init__(self, experiment=None, parameters=None, pv_name='pv_output_linear_elasticity'):
+        # generate "dummy" experiment when none is passed
+        if experiment is None:
+            experiment = experimental_setups.MinimalCubeExperiment(parameters)
+
+        super().__init__(experiment, parameters, pv_name)
+
+    def setup(self):
+        # setup initial temperature material parameters
+        default_p = Parameters()
+        # polynomial degree
+        default_p['degree'] = 2  # default boundary setting
+
+        # parameters for mechanics problem
+        default_p['E'] = None  # Young's Modulus
+        default_p['nu'] = None  # Poisson's Ratio
+        default_p['mu'] = None
+        default_p['lmbda'] = None
+
+        self.p = default_p + self.p
+
+        # expecting E and nu to compute mu and lambda, however you can directly supply mu and lambda
+        # compute material parameters
+        if self.p.mu is None or self.p.lmbda is None:
+            assert self.p.E is not None and self.p.nu is not None
+            self.p.mu = self.p.E / (2.0 * (1.0 + self.p.nu))
+            self.p.lmbda = self.p.E * self.p.nu / ((1.0 + self.p.nu) * (1.0 - 2.0 * self.p.nu))
+
+        # initialize possible paraview output
+        self.pv_file = df.XDMFFile(self.pv_name + '.xdmf')
+        self.pv_file.parameters["flush_output"] = True
+        self.pv_file.parameters["functions_share_mesh"] = True
+
+        # define function space ets.
+        self.V = df.VectorFunctionSpace(self.experiment.mesh, "Lagrange", self.p.degree)  # 2 for quadratic elements
+
+        self.residual = None  # initialize residual
+        # Define variational problem
+        u_trial = df.TrialFunction(self.V)
+        v = df.TestFunction(self.V)
+        self.a = df.inner(self.sigma(u_trial), df.grad(v)) * df.dx
+
+        if self.p.dim == 2:
+            f = df.Constant((0, 0))
+        elif self.p.dim == 3:
+            f = df.Constant((0, 0, 0))
+        else:
+            raise Exception(f'wrong dimension {self.p.dim} for problem setup')
+
+        self.L = df.inner(f, v) * df.dx
+
+        # boundary conditions only after function space
+        self.bcs = self.experiment.create_displ_bcs(self.V)
+
+        # displacement field
+        self.displacement = df.Function(self.V)
+        # TODO better names!!!!
+        self.visu_space_T = df.TensorFunctionSpace(self.experiment.mesh, "Lagrange", self.p.degree)
+
+    # Stress computation for linear elastic problem
+    def sigma(self, v):
+        # v is the displacement field
+        return 2.0 * self.p.mu * df.sym(df.grad(v)) + self.p.lmbda * df.tr(df.sym(df.grad(v))) * df.Identity(len(v))
+
+    def solve(self, t=1.0):
```

Comment:
That is currently not intended that way. 
The steps are defined externally. For this example this is done in the forwardmodel.

Reply:
In order to extend this to nonlinear models with history data, I guess this would be advantageous. In addition, does that mean that each sensor stores all time steps (stacking up all calls to the model), while the model in itself does not now what was computed in that run?

Reply:
I am not sure, it probably depends who you would like to work with the model.
I fell like the time loop should be external, but I guess that is just personal preference. You could build a time stepping class or something if you really wanted to, to which you can feed the problem.
the sensor stores whatever the sensor is measuring for each time step. the problem object has all the relevant data, which could include history data if necessary, as for example my hydration model does.